### ISSUE TITLE
`no-duplicate-hooks`: plain hook calls (e.g. `beforeAll()`) all share the same empty string key, causing false positives

### ISSUE THREAD

**Issue by KristjanESPERANTO** [playwright-community/eslint-plugin-playwright#443]
**Bug Report**

When using `eslint-plugin-playwright` together with a Vitest-based test runner (where e2e tests call `beforeAll()`/`afterAll()` directly, not as `test.beforeAll()`), the rule `no-duplicate-hooks` incorrectly reports the second hook in a describe block as a duplicate – even when they are different hook types.

**Root cause**

In the rule implementation, the hook name is resolved as:
```js
const name = node.callee.type === "MemberExpression"
  ? getStringValue(node.callee.property)
  : "";
```
For plain (non-member-expression) calls the name always falls back to `""`. All hooks – `beforeAll`, `afterAll`, `beforeEach`, `afterEach` – then share the same counter under the key `""`, so the second hook in any describe block is flagged.

**Expected behavior**

Plain calls should be tracked by their actual function name (`node.callee.name`), so `beforeAll` and `afterAll` don't collide:
```js
const name = node.callee.type === "MemberExpression"
  ? getStringValue(node.callee.property)
  : node.callee.name ?? "";
```

**Minimal repro**
```js
// vitest + playwright-plugin: both hooks are flagged as the second is "duplicate"
describe("foo", () => {
  beforeAll(() => {});  // key: ""  → counter[""] = 1
  afterAll(() => {});   // key: ""  → counter[""] = 2 → false positive!
});
```

**Environment**
- `eslint-plugin-playwright`: 2.9.0
- Test runner: Vitest (hooks called as plain `beforeAll()`/`afterAll()`)
